src/main/scala/org/apache/spark/sql/redis/RedisSourceRelation.scala

```diff
@@ -286,4 +280,16 @@ object RedisSourceRelation {
   def uuid(): String = UUID.randomUUID().toString.replace("-", "")
 
   def tableDataKeyPattern(tableName: String): String = s"$tableName:*"
+
+  def tableKey(keysPrefixPattern: String, redisKey: String): String = {
+    if (keysPrefixPattern.isEmpty) {
+      redisKey
+    } else if (keysPrefixPattern.endsWith(":*")) {
```

Comment:
Can we use regex to extract an actual key value from the key? In this case we don't need to put any limitations on the key prefix pattern.

Reply:
Right, let's make a limitation that a pattern should end with an asterisk, but the delimiter could be any char/string. So the pattern will be "xxxxxxx*".

Reply:
So the pattern could be "user:*" or "user#*" or 'user/*'

Reply:
Agree, the check for `:*` is redundant, because `.endsWith("*")` is enough